### PR TITLE
Matched func_8012EF0C and correct D_801C2380

### PR DESCRIPTION
**Pull request by EllipticEllipsis**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Matched our only current NON_EQUIVALENT, corrected the format of the data it uses too: makes somewhat more sense now. Didn't document it since the file's only half-finished. Thanks to @AngheloAlf and the permuter for the last couple of tricks required... that variable's a doozy.

Formatter made a bit off a mess of the data and there's no obvious way to fix it, we may have to consider something drastic.

### CODE REVIEW

File: src/code/code_0x8012EC80.c [hunk 1]
```diff
@@ -229,64 +229,42 @@ s32 func_8012EE34(s16 arg0) {
...
-    {
-        SCENE_20SICHITAI,
-        SCENE_20SICHITAI2,
-        SCENE_21MITURINMAE,
-        SCENE_22DEKUCITY,
-        SCENE_DEKU_KING,
-        SCENE_KINSTA1,
-    },
-    { -1 },
+    { SCENE_20SICHITAI, SCENE_20SICHITAI2, SCENE_21MITURINMAE, SCENE_22DEKUCITY, SCENE_DEKU_KING, SCENE_KINSTA1, -1 },
```
Comment (AngheloAlf) [inline src/code/code_0x8012EC80.c:246]: Formatter did something weird here. Maybe try removing some trailing commas and re-run?

File: src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c [hunk 2]
```diff
@@ -278,7 +278,8 @@ void EnBji01_DialogueHandler(EnBji01* this, GlobalContext* globalCtx) {
                     case 0x5F7:
                     case 0x5F8:
                         func_801477B4(globalCtx);
-                        this->actor.flags &= ~0x10000;;
+                        this->actor.flags &= ~0x10000;
+                        ;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:282]:
```suggestion
                        this->actor.flags &= ~0x10000;
```

File: src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c [hunk 3]
```diff
@@ -287,7 +288,8 @@ void EnBji01_DialogueHandler(EnBji01* this, GlobalContext* globalCtx) {
             break;
         case 6:
             this->actor.params = ENBJI01_PARAMS_FINISHED_CONVERSATION;
-            this->actor.flags &= ~0x10000;;
+            this->actor.flags &= ~0x10000;
+            ;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:292]:
```suggestion
            this->actor.flags &= ~0x10000;
```

File: src/code/code_0x8012EC80.c (hunk 1 above)
Comment (EllipticEllipsis) [reply src/code/code_0x8012EC80.c:246]: I can try again, although first try didn't seem to help.

**Review by AngheloAlf — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]

**Pull request merged**